src/main/java/com/exadel/discountwebapp/location/controller/LocationController.java

```diff
@@ -0,0 +1,48 @@
+package com.exadel.discountwebapp.location.controller;
+
+import com.exadel.discountwebapp.location.entity.Location;
+import com.exadel.discountwebapp.location.service.LocationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/location")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class LocationController {
+    LocationService locationService;
+
+    @GetMapping("/")
+    private List<Location> getAllLocations() {
+        return locationService.findAll();
+    }
+
+    @GetMapping("/{county}")
+    private List<Location> getALlLocationByCounty(@PathVariable String county) {
+        return locationService.findAllByCounty(county);
+    }
+
+    @GetMapping("/{city}")
+    private List<Location> getAllLocationByCity(@PathVariable String city) {
+        return locationService.findAllByCity(city);
+    }
+
+    @PostMapping
+    private Location save(@RequestBody Location location) {
```

Comment:
You should receive the LocationRequestVO from the client

Reply:
done